### PR TITLE
Names and Triggers for Actions

### LINKED ISSUE #169: [ISSUE]: Names and Triggers for Actions
### What is the problem?

- [x] dev-build on push to develop and main
- [x] Better workflow names

### PR DESCRIPTION
**Pull request by Til7701**
## What type of PR is this? (keep all applicable)

- Feature
- Refactor

## Description
 - Actions and steps have descriptive names.
 - There are no redundant project versions anymore. Whenever the project version in not given in the build process, the version is now read from the Maven pom.
 - The discrepancies in project version and build version have been removed. If the given build version is not the same as the project version, Maven is informed about the change and the build version is used to build the jars.

## Related Tickets & Documents

- Closes #169

### CODE REVIEW

File: .github/workflows/build-jar.yml [hunk 1]
```diff
@@ -1,20 +1,25 @@
-name: Build jar with dependencies
+name: Build jar with Dependencies
```
Comment (JayPi4c) [inline .github/workflows/build-jar.yml:1]: Why Dependencies with capital D?

**Review by JayPi4c — APPROVED** [review]
Your choice for upper and lower case letters is interesting but okay.

File: .github/workflows/build-jar.yml (hunk 1 above)
Comment (Til7701) [reply .github/workflows/build-jar.yml:1]: I don't like english capitalization rules. That's why. Most of these are no real sentences. Sometimes nouns are capitalized in english, if it is a headline. Are these headlines or sentences? Or something different? I don't know.

Comment (JayPi4c) [reply .github/workflows/build-jar.yml:1]: Doesn't matter. I need it to be merged to continue with the ppa issue.

**Pull request merged**